service/create/domains_test.go

```diff
@@ -0,0 +1,81 @@
+package create
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/clustertpr"
+	"github.com/giantswarm/clustertpr/kubernetes"
+	"github.com/giantswarm/clustertpr/kubernetes/api"
+	"github.com/juju/errgo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHostedZoneName(t *testing.T) {
+	tests := []struct {
+		name string
+		tpo  awstpr.CustomObject
+		res  string
+		err  error
+	}{
+		{
+			name: "well-formed domain",
```

Comment:
I like that. I would have called it description as it seems to describe the test case, but whatever.